### PR TITLE
DSDEEPB-1714: store the access token in a cookie during login

### PR DESCRIPTION
**Pull request by davidangb**
This is important for download proxy functionality; our rewrite rules look for a cookie.

I value any feedback on this, especially security-related feedback. If we think it's a bad idea to store the token in a cookie, we can come up with another approach - perhaps sending it in the url for download requests (and only for download requests)

### CODE REVIEW

**Comment by dmohs** [comment]
My preference would be to create this cookie only when a user clicks a download link in the UI, and with a very short (5 minute) expiration.

**Comment by rushtong** [comment]
I agree with David on this when the cookie should get triggered.

File: src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala [hunk 1]
```diff
@@ -192,7 +192,13 @@ trait OAuthService extends HttpService with PerRequestCreator with FireCloudDire
       .withPath(Uri.Path./)
       .withFragment(s"$redirectFragment?access_token=$accessToken")
 
-    redirect(Uri(uiRedirect.toString), StatusCodes.TemporaryRedirect)
+    // we require a spray DateTime, not a joda DateTime
+    // expire in one day
+    val expiration = spray.http.DateTime.now + (1000 * 60 * 60 * 24)
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala]: If we don't trigger the cookie on the actual download endpoint, then I think this duration should be shortened. I think the current login expires after an hour - but I can't pin down where that is configured.

Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/service/OAuthService.scala]: Shortening to an hour is fine. I think you're looking for this code: https://github.com/broadinstitute/firecloud-orchestration/blob/develop/src/main/resources/swagger/auth-page.html

If we only place the cookie upon download, we can make it even shorter. I'll work on that.

**Comment by davidangb** [comment]
Okay, please see https://github.com/broadinstitute/firecloud-ui/pull/293. If that PR goes through, this entire PR can be deleted; all the changes are in the UI.

**Comment by davidangb** [comment]
this is now obsolete, given  broadinstitute/firecloud-ui#293. closing.

**Pull request closed without merging**